service/storage_service.cc

```diff
@@ -1433,6 +1447,35 @@ class topology_coordinator {
                     node = std::move(f).get();
                 }
 
+                if (_group0.is_member(node.id, true)) {
+                    // If we remove a node, we make it a non-voter early to improve availability in some situations.
+                    // There is no downside to it because the removed node is already considered dead by us.
+                    //
+                    // FIXME: removenode may be aborted and the already dead node can be resurrected. We should consider
+                    // restoring its voter state on the recovery path.
+                    if (node.rs->state == node_state::removing) {
+                        co_await _group0.make_nonvoter(node.id);
+                    }
+
+                    // If we decommission a node when the number of nodes is even, we make it a non-voter early.
```

Comment:
Why did you decide to do it only for even quorums?

Reply:
In the case of `decommission`, the decommissioning node can be a part of a voting majority. In such a situation, making it a non-voter early would decrease the size of the majority by 1 and that could reduce availability. If the number of nodes is odd, it can happen - 3 nodes, 1 dead, 1 decommissioning. If the number of nodes is even, it can't because the required size of a majority also decreases by 1.

I think the highlighted comment should be more precise and directly answer your question.

Reply:
What's the issue of dropping non-voting rights a bit earlier, after all, we're decommissioning the node anyway?

Reply:
That sounds right.

@kbr-scylla what do you think? Maybe you had something else in mind when mentioning availability reduction for an odd number of nodes in #13959?

Reply:
> What's the issue of dropping voting rights a bit earlier, after all, we're decommissioning the node anyway and the node will be eventually removed from the cluster? 

Not necessarily, the operation may fail and we return the node back to normal.

It's better for availability to not make it a non-voter early with odd number of nodes.

Consider 3 nodes A,B,C, we're decommissioning A. In the middle of streaming, node B fails. We want to abort the operation and return back to normal so we can remove/replace B.

If we made A a non-voter before streaming, we completely lost availability.
But if A is still a voter, we're fine and we can remove/replace B.

Reply:
It is a pretty extreme scenario  - I know of no real world scenario when we want to decommission 1 node out of 3. If anything, we add nodes to such a small cluster. 
A simpler rule would be to not make a nonde a non-voter early in a 3 node cluster, for the reasons you described above, it will reduce the applicability of this rule to exactly the scenario you described above.